sympy/solvers/solveset.py

```diff
@@ -934,20 +933,19 @@ def solveset(f, symbol=None, domain=S.Complexes):
             raise NotImplementedError(filldedent('''
                 relationship between value and 0 is unknown: %s''' % b))
 
-    reps = OrderedDict(zip(
+    reps = list(zip(
         # do larger expressions first in case they contain
         # other Abs
         reversed(list(ordered(
             [i for i in f.atoms(Abs) if symbol in i.free_symbols]))),
         numbered_symbols("t", cls=Dummy)))
     for k in reps:
-        f = f.replace(k, reps[k])
+        f = f.replace(k[0], k[1])
```

Comment:
oops -- thanks.

I might also recommend that you either traverse the elements as:

```
for o, n in reps:
    f = f.replace(o, n)
```
or simply write `f = f.replace(*k)` which unpacks the tuple for you.

It also occurs to me that the arg of Abs may contain things that should be rewritten as Piecewise -- will that happen when the Abs is handled?

Reply:
@smichr For cases with Abs having args that should be rewritten into `Piecewise` like:
```
>>>solveset(Abs(Max(x, 2) +4*x)- 33, x, S.Reals)            # for this branch
Union(ConditionSet(x, Eq(4*x + Max(2, x) - 33, 0), S.Reals), \
ConditionSet(x, Eq(4*x + Max(2, x) + 33, 0), S.Reals))
# same as the current master
```
While the correct answer should be:
(which comes when it is rewitten into Piecewise)
```
>>>solveset(Abs(Max(x, 2) +4*x)- 33, x, S.Reals)   
{-35/4, 33/5}
```